paimon-common/src/main/java/org/apache/paimon/casting/StringToMapCastRule.java

```diff
@@ -176,7 +176,7 @@ private Object parseValue(String valueStr, CastExecutor<BinaryString, Object> ca
                 : castExecutor.cast(BinaryString.fromString(valueStr));
     }
 
-    private List<String> splitMapEntries(String content) {
+    public List<String> splitMapEntries(String content) {
```

Comment:
Maybe we can introduce something like Spark `CollationAwareUTF8String.splitSQL`.

Reply:
Let's research it, but can we deal with it with this tiny change?